src/server/components/delete-modal.jsx

```diff
@@ -4,7 +4,7 @@ import ModalTitle from "../../shared/components/modal-title";
 import ModalContent from "../../shared/components/modal-content";
 import ModalCall from "../../shared/components/modal-call";
 import { TextButton } from "../../shared/components/buttons";
-import { fetchWithCSRFTokenAndJSONContent } from "../../shared/fetch-with-csrf-token";
+import { deleteFileOnServer } from "../../shared/utils/file-operations";
```

Comment:
we need to start grouping all of our server API functions in one place rather than dumping them in `utils` with a bunch of other unrelated stuff. this should go in a folder called `server-api` or `server-access-functions` or whatever you wish

Reply:
also, don't import the API function `deleteFileOnServer` into the component -- the component should dispatch a thunk that is passed into it via mapStateToProps, b/c that will make the component more encapsulated and more easily testable

Reply:
cc @wlach -- lets make sure that going forward we start gathering all of the code that talks to the Iodide server API in that location rather than sprinkling it throughout the codebase. the architecture we want to aim for is:

- helper functions that wrap details of the API calls and expose the specific functionality available from the server
- redux thunks that call the API wrappers
- components that dispatch redux thunks

(if this does not make sense to any of you, happy to talk about it further)

Reply:
This has nothing to do with redux – it's a server React component that doesn't use thunks.  Both the server and the client both need the exact same functions like `deleteFileOnServer`, which was the point of this PR.

Reply:
oh my apologies, disregard everything i said in this and the other in-line comment, and sorry for the distraction -- this is all looks good for non-notebook stuff.

the overall architectural direction still applies, though, so it'd still be better to have the API helper functions moved to their own folder in /shared rather than putting them in /shared/utils with other unrelated stuff. we'll want to move a bunch of the existing server API helper functions out of /editor and consolidate them in one place, and the amount of code we'll have is worthy of it's own folder rather than a bunch of files in /utils